pkg/actions/addon/create.go

```diff
@@ -179,13 +173,22 @@ func (a *Manager) patchAWSNodeDaemonSet() error {
 
 	return nil
 }
-func (a *Manager) getRecommendedPolicies(addon *api.Addon) []string {
+
+func (a *Manager) hasRecommendedPolicies(addon *api.Addon) bool {
+	updatedAddon := addon.DeepCopy()
+	a.setRecommendedPolicies(updatedAddon)
+	return hasPoliciesSet(updatedAddon)
+}
```

Comment:
this felt a bit odd writing it, going to comeback to it as I don't like this.

`setRecommendedPolicies` mutates the `addon` that is passed in. Sometimes we just want to check if an addon has any recommended policies, without mutating the object, so this approach works. but it feels wrong :joy:

Reply:
I would re-write the whole thing given this opportunity. The side effect of this is super weird and having to deepCopy this thing is also weird. In general eksctl is full of side effects and functions doing weird shit. :D Is refactoring it a too big of a work?

Reply:
its not too big, I wrote this ages ago so don't quite remember but I definitely dislike the current behaviour